apis/k8ssandra/v1alpha1/k8ssandracluster_types.go

```diff
@@ -277,38 +287,68 @@ type EmbeddedObjectMeta struct {
 // TODO Implement Stringer interface. It will helpful for debugging and testing.
 type CassandraConfig struct {
 	// +optional
-	CassandraYaml *CassandraYaml `json:"cassandraYaml,omitempty"`
+	CassandraYaml CassandraYaml `json:"cassandraYaml,omitempty"`
```

Comment:
@jsanda just to make sure you are on board here: I realized that both `CassandraYaml` and `JvmOptions` always end up non-nil due to a lot of default entries being applied during reconciliation, so I figured it would be much easier to remove the pointers altogether.

Reply:
That works for me provided it doesn't requiring the user to define something for those fields.

Reply:
Everything inside CassandraYaml and JvmOptions is a pointer, so I think we are good.